suites/apis/googleServiceAccountRemovalTest.js

```diff
@@ -0,0 +1,471 @@
+const { Commons } = require('../../utils/commons.js');
+
+
+Feature('GoogleServiceAccountRemoval');
+
+/**
+ * Test that the SA removal job, fence-create google-manage-user-registrations,
+ * deletes invalid SA and Google projects.
+ *
+ * To check that the user does not have access to the data anymore after the
+ * job cleans up a SA, we need to create a key before the clean up, and make
+ * sure we cannot use it after the clean up. Since the validation checks that
+ * there is no external access to the SA, creating a key would prevent us from
+ * testing other removal use cases. This is why the data access is only
+ * checked in the "SA has external access" test.
+ */
+
+
+BeforeSuite(async (google, fence, users) => {
+  await fence.complete.suiteCleanup(google, users);
+});
+
+
+After(async (google, fence, users) => {
+  await fence.complete.suiteCleanup(google, users);
+});
+
+
+AfterSuite(async (google, fence) => {
+  await google.unlockGoogleProject(fence.props.googleProjectDynamic);
+});
+
+
+function checkAndCleanSA() {
+  var fenceCmd = 'fence-create --verbose google-manage-user-registrations';
+  console.log(`Running: ${fenceCmd}`);
+  var jobRes = bash.runCommand(fenceCmd, 'fence');
+  console.log(jobRes);
+  return jobRes;
+}
+
+
+Scenario('SA removal job test: no access removal when SA is valid @reqGoogle', async (fence, users, google, files) => {
+  // test that the clean up job does not remove access to valid SA/projects
+
+  // Setup
+  const googleProject = fence.props.googleProjectDynamic;
+  await fence.complete.forceLinkGoogleAcct(users.user0, googleProject.owner);
+
+  // Register account
+  let registerRes = await fence.do.registerGoogleServiceAccount(
+    users.user0,
+    googleProject,
+    ['QA']
+  );
+  fence.ask.responsesEqual(registerRes, fence.props.resRegisterServiceAccountSuccess);
+
+  // Lock the project
+  lockRes = await google.lockGoogleProject(googleProject);
+  chai.expect(lockRes, 'Could not lock project').to.be.true;
+
+  // Get creds to access data
+  let [pathToKeyFile, keyFullName] = await google.createServiceAccountKeyFile(googleProject);
+
+  // Access data
+  user0AccessQARes = await google.getFileFromBucket(
+    fence.props.googleBucketInfo.QA.googleProjectId,
+    pathToKeyFile,
+    fence.props.googleBucketInfo.QA.bucketId,
+    fence.props.googleBucketInfo.QA.fileName
+  );
+
+  // Delete the creds so that the clean up job will not mark this SA as invalid
+  await google.deleteServiceAccountKey(keyFullName);
+  files.deleteFile(pathToKeyFile);
+
+  // Run clean up job
+  checkAndCleanSA();
+
+  // Get creds to access data
+  [pathToKeyFile, keyFullName] = await google.createServiceAccountKeyFile(googleProject);
+
+  // Access data
+  user0AccessQAResAfter = await google.getFileFromBucket(
+    fence.props.googleBucketInfo.QA.googleProjectId,
+    pathToKeyFile,
+    fence.props.googleBucketInfo.QA.bucketId,
+    fence.props.googleBucketInfo.QA.fileName
+  );
+
+  // Clean up
```

Comment:
this won't happen if this fails before this on the `fence.ask` or `expect` for locking. will that cause an issue for a later test env? the SA's are per env now, so that shouldn't matter. but I'm worried about the linked google accounts which are shared. are you sure that every test removes all linked accounts before trying anything?  Or are you sure that already having an account linked won't break other tests?

Reply:
so the db won't have the google account linking information in another env b/c we wipe the db. but the actual google account may still be in the proxy group.. I'm trying to think through if this is a problem

Reply:
actually, I think this is fine... b/c when we do SA validation, to check linkage we check the fence db *not* Google. and we don't have any tests for accessing data with the linked account

Reply:
i will change this to lock before registering the SA like you suggested, so if `expect(lock)` fails there will be no SA creation so nothing to delete. for the linked google account, it should be unlinked by the suite clean up